starknet_devnet/starknet_wrapper.py

```diff
@@ -939,3 +942,71 @@ async def is_deployed(self, address: int) -> bool:
         cached_state = self.get_state().state
         class_hash = await cached_state.get_class_hash_at(address)
         return bool(class_hash)
+
+    async def abort_blocks(self, starting_block: StarknetBlock) -> str:
+        """
+        Abort blocks.
+        """
+        # Check if genesis block can be aborted.
+        if starting_block.block_number == self.genesis_block_number:
+            raise StarknetDevnetException(
+                code=StarknetErrorCode.OUT_OF_RANGE_BLOCK_ID,
+                message="Aborting genesis block is not supported.",
+            )
+
+        # Check if blocks can be aborted in fork mode.
+        if (
+            self.config.fork_block
+            and self.config.fork_block >= starting_block.block_number
+        ):
+            raise StarknetDevnetException(
+                code=StarknetErrorCode.OUT_OF_RANGE_BLOCK_ID,
+                message="Aborting forked blocks is not supported.",
+            )
+
+        # Create new block with pending transactions if possible.
+        # We need to store them so later we can change the status to REJECTED.
+        if self.blocks.is_block_pending():
+            await self.generate_latest_block()
+
+        aborted_blocks = []
+        current_block = await self.blocks.get_last_block()
+
+        # Before the while loop to abort blocks, check if block numbers are set.
+        if not (current_block.block_number and starting_block.block_number):
+            raise StarknetDevnetException(
+                code=StarknetErrorCode.BLOCK_NOT_FOUND,
+                status_code=400,
+                message="Aborting blocks must have block numbers. This can happen during abortion of the same block twice.",
```

Comment:
I think blocks needing a block number is an implementation detail.
```suggestion
                message="Block cannot be aborted. Make sure you are aborting an accepted block.",
```

Reply:
Fixed in 83173ec140027508606643845f52369c1f5da49e